pixelator.html

```diff
@@ -1,126 +1,172 @@
+      margin-right: .5em;
+    }
+
+    #graphics {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+    }
+  </style>
+</head>
+
+<body onload="initPage()">
+<div id="controls">
+  <div class="input-group">
+    <label for="imageFile">Image File:</label>
+    <input type="file" name="" id="imageFile">
+  </div>
+  <div class="input-group">
+    <label for="multiplierInput">Pixelation Factor:</label>
+    <input type="number" id="multiplierInput" min="0" max="666" value="2" />
+  </div>
+</div>
+<div id="graphics">
+  <img id="sourceImage" src="" alt="source image"/>
+  <canvas id="targetCanvas"></canvas>
+</div>
+Joel Peltonen 2021
+
+<script type="text/javascript">
+
+  const initialFilename = 'Pink-Gonzo.jpg';
+  const maxFileSize = 900000;
+  const maxPixelationFactor = 666;
+  const minPixelationFactor = 2;
+  const initialPixelationFactor = 5;
+  let pixelationFactor = initialPixelationFactor;
+
+  // Are you KIDDING me with this type casting?
+  const canvas = /** @type {HTMLCanvasElement} */ document.querySelector('#targetCanvas');
+  const ctx = /** @type {CanvasRenderingContext2d} */ canvas.getContext('2d');
+  const fileDialog = document.querySelector("#imageFile");
+  const sourceImage = document.querySelector('#sourceImage');
+  const multiplierInput = document.querySelector("#multiplierInput");
+
+  // init page (triggered by body onload event)
+  function initPage() {
+    sourceImage.setAttribute('src', initialFilename);
+    multiplierInput.setAttribute('value', pixelationFactor);
+  }
+
+  // event handlers
+  fileDialog.addEventListener('change', () => {
+    readFile();
+    draw();
+  })
+
+  multiplierInput.addEventListener('change', () => {
+    pixelationFactor = parseInt(multiplierInput.value);
+    if (pixelationFactor > maxPixelationFactor) {
+      multiplierInput.value = maxPixelationFactor;
+      return alert(`Pixelation factor exceeds maximum of ${maxPixelationFactor}!`);
+    } else if (pixelationFactor < minPixelationFactor) {
+      multiplierInput.value = minPixelationFactor;
+      return alert(`Pixelation factor falls below minimum of ${minPixelationFactor}!`);
+    }
+    draw();
+  })
+
+  sourceImage.addEventListener('load', () => {
+    draw(); // triggered when sourceImage was loaded
+  });
+
+  function draw() {
+    let imageHeight = sourceImage.height;
+    let imageWidth = sourceImage.width;
+    canvas.width = imageWidth;
+    canvas.height = imageHeight;
+
+    // draw centered source image
+    const posX = Math.ceil(canvas.width/2 - imageWidth/2);
+    const posY = Math.ceil(canvas.height/2 - imageHeight/2);
+    ctx.drawImage(sourceImage, posX, posY, imageWidth, imageHeight);
+
+    // read rendered image pixels
+    const imageData = ctx.getImageData(posX, posY, imageWidth, imageHeight);
+    const pixelvalues = imageData.data;
+    // console.log(pixelvalues)
+
+    // draw pixelized image
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
```

Comment:
Ah, I see this is a different version from the one you currently have where you have original and the pixelated one side by side!